pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ComparisonColumns.java

```diff
@@ -37,10 +38,45 @@ public int getComparableIndex() {
     return _comparableIndex;
   }
 
+  public int compareToSealed(ComparisonColumns other) {
+      /*
+       - iterate over all columns
+       - if any value in _values is greater than its counterpart in _other._values, keep _values as-is and return 1
+       - if any value in _values is less than its counterpart  in _other._values _and_ none are greater than their
+         counterpart in _other._values, keep _values as-is and return -1
+           - i.e. no value in _this_ is greater than its counterpart in _other
```

Comment:
I don't follow this part. If any value in _values is less than its counterpart, we should just return -1. No need to check other values

Reply:
The original goal was to ensure that a default value like Long.MIN_VALUE stored for a comparison column would not cause accidental rejection of data.  But in looking at this again after some time, I agree with your assessment that it seems sufficient to simply return if any value is less that its counterpart.  If the values were both null i.e. default value, the `compareTo` would return `0` so that case would be handled without issue.

Reply:
made this change as of [634b4cb](https://github.com/apache/pinot/pull/10704/commits/634b4cb2b26831bfa4204a73372cc0cf49bc8147)